test/js/node/http/node-http.test.ts

```diff
@@ -1746,3 +1746,37 @@ if (process.platform !== "win32") {
     expect([joinPath(import.meta.dir, "node-http-ref-fixture.js")]).toRun();
   });
 }
+
+it("#10177 response.write with non-ascii latin1 should not cause duplicated character or segfault", async () => {
+  const expected = "á" + "-".repeat(254) + "x";
+  const server = require("http")
+    .createServer((_, response) => {
+      response.write(expected);
+      response.write("");
+      response.end();
+    })
+    .listen(0, "localhost", async (err, hostname, port) => {
+      try {
+        expect(err).toBeFalsy();
+        expect(port).toBeGreaterThan(0);
+        const url = `http://${hostname}:${port}`;
+        const count = 1000;
```

Comment:
I think we can safely reduce the count if we add iterations above, probably 20 is fine

Reply:
Do we even need to repeat same tests? It seems like fuzzing to me. If we test for coincidences, errors should be caught by repeating such tests in total or by them just being run hundreds of times in future, I think.

Reply:
We want to check that repeatedly allocating memory of slightly different sizes and types produces the correct results every time. GC + network adds a lot of unpredictability so it helps to run the same or similar code repeatedly